url/src/quirks.rs

```diff
@@ -141,7 +141,7 @@ pub fn set_host(url: &mut Url, new_host: &str) -> Result<(), ()> {
     if host == Host::Domain("".to_string()) {
         if !username(&url).is_empty() {
             return Err(());
-        } else if let Some(Some(_)) = opt_port {
+        } else if let Some((_, Some(_))) = opt_port {
```

Comment:
This might be considered a breaking change? I consider checking for user input regardless of defaults "more" correct, this might have been considered a bug until now.
This can easily be removed.